### DISCUSSION TITLE
Toggleables can't toggle independently?

### DISCUSSION THREAD

**Discussion by matt17r** [excid3/tailwindcss-stimulus-components#250] [Ideas]
I'm trying to use `Toggle` to switch which icon is hidden in a button when the button is clicked. Even if I start with Icon A visible and Icon B hidden, toggle synchronises them on first run.

Does `Toggle` support toggling two items with opposing values? If not, is this a valid use case that's worth adding support for?

    <div data-controller="toggle" class="-mr-2 flex md:hidden">
      <button data-action="click->toggle#toggle touch->toggle#toggle" type="button" class="relative inline-flex items-center justify-center rounded-md bg-blue-900 p-2 text-blue-50">
        
        <svg data-toggle-target="toggleable" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5" />
        </svg>
        
        <svg data-toggle-target="toggleable" class="hidden h-6 w-6" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor">
          <path stroke-linecap="round" stroke-linejoin="round" d="M6 18L18 6M6 6l12 12" />
        </svg>
      </button>
    </div>

**Comment by excid3** [comment]
Toggle's only designed for a single element currently.

**Comment by jaykilleen** [comment]
Hmm interesting.

I am looking at this now and seeing if I can make something similar work. In my case, I have a search form with simple fields and advanced fields. Similar to the example in this dribble. 

Essentially I want:
- double down arrow ⬇️⬇️ is clicked
- advanced search fields are **shown**
- double down arrow is **hidden**
- double up arrow is **shown** ⬆️⬆️
- double up arrow is clicked
- advanced search fields are **hidden**
- double down arrow ⬇️⬇️ is **shown**

https://dribbble.com/shots/13634106-Admintory-Order-Details
![image](https://github.com/user-attachments/assets/93aa82ec-68d0-413b-8e2e-259d463b56f1)

Whilst toggle can only have one element, could toggle and dropdown be used together so that one click executes two events?

**Reply by jaykilleen** [reply]
OK this is what I have come up with with the limited time I have.

I have kept it simple for anyone else trying to just get this done. 
No transitions, no fancy stuff. No extending existing controllers. Just Stimulus.

Now I have a nice little button that toggles the extra fields below.

![image](https://github.com/user-attachments/assets/2de879af-0bd6-43e3-86c1-9b69210f57ca)
![image](https://github.com/user-attachments/assets/4d83b6c1-ab40-42b3-9659-6f64229dc421)


**switch_controller.js**
```
import { Controller } from '@hotwired/stimulus'

export default class extends Controller {
  static targets = ['switchable']

  toggle() {
    this.switchableTargets.forEach(target => {
      target.classList.toggle('hidden')
    })
  }
}

```

then in my view I use a combination of the dropdown and switch controllers to change the buttons.

```
<div  data-controller="dropdown switch"
      id="search-menu"
      class="flex flex-col" >
  <div id="basic-search-menu" class="flex flex-col md:flex-row w-full mt-2 gap-2 items-center">
    <!-- all the basic search fields -->
    <button id="advanced-search-expand-button"
            type="button"
            data-switch-target='switchable'
            data-action="click->dropdown#toggle 
                          keydown.esc->dropdown#hide
                          click->switch#toggle"
            data-turbo="false"
            class="flex items-center justify-center rounded-md border-0 p-1.5 text-gray-500 ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-sapphire sm:text-sm/6">
      <%= heroicon "chevron-double-down", variant: :outline, options: { class: "flex-shrink-0 size-6" } %>
    </button>
    <button id="advanced-search-collapse-button"
            type="button"
            data-switch-target='switchable'
            data-action="click->dropdown#toggle 
                          keydown.esc->dropdown#hide
                          click->switch#toggle"
            data-turbo="false"
            class="hidden flex items-center justify-center rounded-md border-0 p-1.5 text-gray-500 ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-sapphire sm:text-sm/6">
      <%= heroicon "chevron-double-up", variant: :outline, options: { class: "flex-shrink-0 size-6" } %>
    </button>
  </div>
  <div  class="hidden" 
        id="advanced-search-menu"
        role="menu"
        data-dropdown-target="menu"
        data-dropdown-open-value="false">
    <!-- all the advanced search fields -->
  </div>
</div>
```

**Comment by jaykilleen** [comment]
Also for anyone else trying to implement this functionality, my current implementation of this all sits inside a form (ransack's `search_form_for` which has `method: :get, data: { turbo: true, turbo_stream: true }`.

What this means is that the buttons clicked to *toggle* each other need to have the `data-turbo="false"` and `type="button"` otherwise they will be treated as submit buttons for the form and then the turbo events will do weird stuff to your tailwindcss-stimulus-controllers. Learnt that lesson the hard way.
